pkg/agent/agent.go

```diff
@@ -175,6 +175,49 @@ type BufferStat struct {
 	Timestamp time.Time
 }
 
+type PGDatabase struct {
+	BlksHit               int64
+	BlksRead              int64
+	TuplesReturned        int64
+	TuplesFetched         int64
+	TuplesInserted        int64
+	TuplesUpdated         int64
+	TuplesDeleted         int64
+	TempFiles             int64
+	TempBytes             int64
+	Deadlocks             int64
+	IdleInTransactionTime float64
+	Timestamp             time.Time
+}
+
+type PGBGWriter struct {
+	BuffersClean    int64
+	MaxWrittenClean int64
+	BuffersAlloc    int64
+	Timestamp       time.Time
+}
+
+type PGWAL struct {
+	WALRecords     int64
+	WALFpi         int64
+	WALBytes       int64
+	WALBuffersFull int64
+	WALWrite       int64
+	WALSync        int64
+	WALWriteTime   float64
+	WALSyncTime    float64
+	Timestamp      time.Time
+}
+
+type PGCheckPointer struct {
+	NumTimed       int64
+	NumReq         int64
+	WriteTime      float64
+	SyncTime       float64
+	BuffersWritten int64
+	Timestamp      time.Time
+}
```

Comment:
Not crucial but if you have doc links to where you got these from, it would be useful to include.